src/components/MovieForm/MovieForm.test.js

```diff
@@ -0,0 +1,202 @@
+import React from "react";
+import MovieForm from "./MovieForm";
+import {render, fireEvent, waitForElement} from '@testing-library/react';
+import {paths} from "../../constants/routes";
+import {createMemoryHistory} from 'history';
+import {MemoryRouter, Router} from "react-router";
+
+jest.mock("react-router-dom", () => ({
+    ...jest.requireActual("react-router-dom"),
+    useParams: () => ({
+        id: 1
+    })
+}));
+
+let reactRouterDom = require("react-router-dom");
+
+describe("MovieForm POST", () => {
+    let postFail;
+    let movieData;
+
+    beforeAll(() => {
+        global.fetch = jest.fn().mockImplementation((input, init) => {
+            return new Promise((resolve, reject) => {
+                if (postFail) {
+                    resolve({status: 500});
+                } else {
+                    resolve({status: 201, json: () => Promise.resolve({})});
+                }
+            });
+        });
+    });
+
+    beforeEach(() => {
+        jest.clearAllMocks();
+        reactRouterDom.useParams = () => ({
+
+        });
+        movieData = {
+            id: 1,
+            movieName: "Interstellar",
+            movieGenre: "Sci-Fi",
+            movieDirector: "Christopher Nolan",
+            movieReleaseDate: new Date(),
+            movieImage: "",
```

Comment:
Nigdzie nie wykorzystujemy w testach movieImage, można dodać to gdzieś, albo stąd usunąć (bo w końcu jak rozumiem obrazek nie jest required dla bazy danych)